File: Elego/random_agent.py
'''
Agent that selects an random move with uniform probability
Used only for debugging and benchmarking purposes
'''
from agent import Agent
from gameplay.gamestate import GameState
from gameplay.board import Board
from gameplay.move import Move
from random import randint

class RandomAgent(Agent):
    def __init__(self):
        pass

    def select_move(self,gamestate):
        legal_moves = gamestate.legal_moves()
        number_of_moves = len(legal_moves)
        index = randint(0,number_of_moves-1)
        return legal_moves[index]

File: SelfplayEngine/agents/random_agent.cc
#include "random_agent.h"
#include <vector>
#include <cstdio>

Point RandomAgent::select_move(GameState gamestate){
    std::vector<Point> legal_moves = gamestate.legal_moves();
    int random_index = rand()%(legal_moves.size());
    Point random_move = legal_moves[random_index];
    
    return random_move;
}